spec/reek/cli/application_spec.rb

```diff
@@ -0,0 +1,33 @@
+require_relative '../../spec_helper'
+require_lib 'reek/cli/application'
+
+RSpec.describe Reek::CLI::Application do
+  describe '#initialize' do
+    it 'exits with default error code on invalid options' do
+      error_raised = false
+      begin
+        Reek::CLI::Silencer.silently do
+          Reek::CLI::Application.new ['--foo']
+        end
+      rescue SystemExit => e
+        expect(e.status).to eq Reek::CLI::Options::DEFAULT_ERROR_EXIT_CODE
+        error_raised = true
+      end
+      expect(error_raised).to be_truthy
+    end
```

Comment:
Hm, could this be written with the exception matcher? Something to the tune of (untested):

```ruby
call = -> { Reek::CLI::Silencer.silently { Reek::CLI::Application.new ['--foo'] } }
expect(call).to raise_error do |error|
 expect(error).to be_a SystemExit
 expect(error.status).to eq Reek::CLI::Options::DEFAULT_ERROR_EXIT_CODE
end
```